interface/patient_file/reminder/patient_reminders.php

```diff
@@ -246,10 +246,9 @@ function sel_patient() {
                     "a.sms_status, a.email_status, a.mail_status, b.fname, b.lname, b.hipaa_allowemail, b.hipaa_allowsms " .
         "FROM `patient_reminders` as a, `patient_data` as b " .
         "WHERE a.active='1' AND a.pid=b.pid " . $add_sql .
-        "ORDER BY " . add_escape_custom($sortby) . " " .
-        add_escape_custom($sortorder) . " " .
-        "LIMIT " . add_escape_custom($begin) . ", " .
-        add_escape_custom($listnumber);
+        "ORDER BY  ? ? "  .
+        "LIMIT ?,?";
```

Comment:
You can not use sql binding on ORDER BY or LIMIT. This code will break. See above for the solution to escape the LIMIT. And we of course have solutions to escape both variable in ORDER BY. The $sort_order can be escaped with the escape_sort_order() function (see https://github.com/openemr/openemr/blob/master/library/formdata.inc.php#L66) which is always available. The $sortby involves a mechanism that we have to escape column names(it does this via whitelisting to only accept table names that actually exist suing this function https://github.com/openemr/openemr/blob/master/library/formdata.inc.php#L85); the ideal solution is dependent on what the possible options for the tables are; if you give me examples of what they can be, I can give you way to escape it.

Reply:
I agree.
Thanks for the explanation, cool functions.

Done.